### PR TITLE
Add auth errors

### PR DESCRIPTION
**Pull request by tschuy**

### CODE REVIEW

File: source/draft_errors.rst [hunk 1]
```diff
@@ -118,3 +118,36 @@ format (e.g. a slug with special characters or a non-numeric ID field).
...
+This error is returned when the username passed for authentication is not a
+valid username.
+
+.. code-block:: javascript
+
+{
+    "status": 401
+    "error": "Invalid username",
+    "text": "${username} is not a valid username"
+}
```
Comment (pop) [inline source/draft_errors.rst]: This block should be indented.

File: source/draft_errors.rst [hunk 2]
```diff
@@ -118,3 +118,36 @@ format (e.g. a slug with special characters or a non-numeric ID field).
...
+the error may change based on what kind of authentication backend the TimeSync
+server is running.
+
+.. code-block:: javascript
+
+{
+"status": 401
+"error": "Authentication failure",
+"text": "Invalid password" / "Bad oAuth token" / etc
+}
```
Comment (pop) [inline source/draft_errors.rst]: Indent this code-block.

File: source/draft_errors.rst [hunk 3]
```diff
@@ -118,3 +118,36 @@ format (e.g. a slug with special characters or a non-numeric ID field).
...
+This error is returned when authentication fails for a valid user. The text of
+the error may change based on what kind of authentication backend the TimeSync
+server is running.
+
+.. code-block:: javascript
+
+    {
+    "status": 401
+    "error": "Authentication failure",
+    "text": "Invalid password" / "Bad oAuth token" / etc
```
Comment (pop) [inline source/draft_errors.rst]: The text in these `{ ... }` should also be indented for consistency.

**Comment by pop** [comment]
Looks good to me.
Minor formatting fixes, once those are resolved: :+1:

**Comment by Kennric** [comment]
This looks like it covers the bases, +1

**Pull request merged**